modules/solid_mechanics/src/materials/IsotropicPlasticity.C

```diff
@@ -40,10 +41,8 @@ IsotropicPlasticity::IsotropicPlasticity(const InputParameters & parameters)
 
     _plastic_strain(declareProperty<SymmTensor>("plastic_strain")),
     _plastic_strain_old(declarePropertyOld<SymmTensor>("plastic_strain")),
-    _scalar_plastic_strain(_hardening_function ? &declareProperty<Real>("scalar_plastic_strain")
-                                               : NULL),
-    _scalar_plastic_strain_old(
-        _hardening_function ? &declarePropertyOld<Real>("scalar_plastic_strain") : NULL),
+    _scalar_plastic_strain(declareProperty<Real>("scalar_plastic_strain")),
```

Comment:
You don't need these.  You have access to `_effective_inelastic_strain` from the base class.